scripts/flow_counters_stat

```diff
@@ -0,0 +1,256 @@
+        data = {ns: {}}
+        if not name_map:
+            return data
+
+        for name, counter_oid in name_map.items():
+            values = self._get_stats_value(counter_oid)
+
+            full_table_id = RATES_TABLE_PREFIX + counter_oid
+            counter_data =  self.db.get(self.db.COUNTERS_DB, full_table_id, PPS_FIELD)
+            values.append(STATUS_NA if counter_data is None else counter_data)
+            values.append(counter_oid)
+            data[ns][name] = values
+        return data
+
+    def _get_stats_value(self, counter_oid):
+        """Get statistic value from COUNTERS_DB COUNTERS table
+
+        Args:
+            counter_oid (string): OID of a generic counter
+
+        Returns:
+            values (list): A list of statistics value
+        """
+        values = []
+        full_table_id = FLOW_COUNTER_TABLE_PREFIX + counter_oid
+        for field in flow_counters_fields:
+            counter_data =  self.db.get(self.db.COUNTERS_DB, full_table_id, field)
+            values.append(STATUS_NA if counter_data is None else counter_data)
+        return values
+
+    def _save(self):
+        """Save flow counter statistic to a file
+        """
+        try:
+            if os.path.exists(self.data_file):
+                os.remove(self.data_file)
+
+            with open(self.data_file, 'wb') as f:
+                pickle.dump(self.data, f)
+        except IOError as e:
+            print('Failed to save statistic - {}'.format(repr(e)))
+
+    def _load(self):
+        """Load flow counter statistic from a file
+
+        Returns:
+            dict: A dictionary. E.g: {<namespace>: {<trap_name>: [<value_in_pkts>, <value_in_bytes>, <rx_pps>, <counter_oid>]}}
+        """
+        if not os.path.exists(self.data_file):
+            return None
+
+        try:
+            with open(self.data_file, 'rb') as f:
+                data = pickle.load(f)
+        except IOError as e:
+            print('Failed to load statistic - {}'.format(repr(e)))
+            return None
+
+        return data
+
+    def _diff(self, old_data, new_data):
+        """Do a diff between new data and old data.
+
+        Args:
+            old_data (dict): E.g: {<namespace>: {<trap_name>: [<value_in_pkts>, <value_in_bytes>, <rx_pps>, <counter_oid>]}}
+            new_data (dict): E.g: {<namespace>: {<trap_name>: [<value_in_pkts>, <value_in_bytes>, <rx_pps>, <counter_oid>]}}
+        """
+        if not old_data:
+            return
+
+        for ns, stats in new_data.items():
+            if ns not in old_data:
+                continue
+            old_stats = old_data[ns]
+            for name, values in stats.items():
+                if name not in old_stats:
+                    continue
+
+                old_values = old_stats[name]
+                if values[-1] != old_values[-1]:
+                    # Counter OID not equal means the trap was removed and added again. Removing a trap would cause
```

Comment:
This is a good logic to handle. However, what if OID returned is the same as before delete? Should we rather not check if the diff is negative and consider it as a case where trap got deleted?